### ISSUE TITLE
Syscall test hangs randomly

### ISSUE THREAD

**Issue by Hsy-Intel** [asterinas/asterinas#1607]
### Describe the bug

During syscall test, there are a low possibility that the hang problems will be triggered. 

1. The `sys_wait4` syscall function:

https://github.com/asterinas/asterinas/blob/85d2c9634f5c4f7b3dac1f612655a5ce675a1422/kernel/src/syscall/wait4.rs#L46

2. The `sys_brk` syscall function:

https://github.com/asterinas/asterinas/blob/e6c613f53841983765a7b3c56ea9958775c76199/kernel/src/syscall/brk.rs#L14
 
These issues occur more often in TDX environment (#1595 case 6, 7).

### To Reproduce

Run `make run AUTO_TEST=syscall` and the test case are: `sendfile_test` and `alarm_test`.

### Logs

1. `sys_wait4` hang log:

```txt
syscall_number: 158  syscall_return: Ok(Return(0))
syscall_number: 218  syscall_return: Ok(Return(276))
syscall_number: 273  syscall_return: Ok(Return(0))
syscall_number: 334  syscall_return: Err(Error { errno: ENOSYS, msg: Some("Syscall was unimplemented") })
syscall_number: 63  syscall_return: Ok(Return(0))
syscall_number: 302  syscall_return: Ok(Return(0))
syscall_number: 89  syscall_return: Ok(Return(16))
syscall_number: 318  syscall_return: Ok(Return(8))
syscall_number: 12  sys_brk 1  new heap end = 10023000sys_brk 2  sys_brk 3  syscall_return: Ok(Return(268578816))
syscall_number: 10  sys_mprotect 1  sys_mprotect 2  sys_mprotect 3  sys_mprotect 4  sys_mprotect 5  sys_mprotect 6
syscall_return: Ok(Return(0))
syscall_number: 157  syscall_return: Ok(Return(0))
syscall_number: 102  syscall_return: Ok(Return(0))
syscall_number: 0  sys_read 1  sys_read 2  sys_read 3  syscall_return: Ok(Return(196))
syscall_number: 1  syscall_return: Ok(Return(196))
syscall_number: 0  sys_read 1  sys_read 2  sys_read 3  syscall_return: Ok(Return(0))
syscall_number: 231  syscall_return: Ok(Return(0))
sys_wait4 8  Printing stack trace:


```

2. `sys_brk` hang log:

```txt
syscall_number: 107  syscall_return: Ok(Return(0))
syscall_number: 102  syscall_return: Ok(Return(0))
syscall_number: 108  syscall_return: Ok(Return(0))
syscall_number: 104  syscall_return: Ok(Return(0))
syscall_number: 158  syscall_return: Err(Error { errno: EINVAL, msg: Some("Invalid enum value") })
syscall_number: 12  sys_brk 1  new heap end = 0sys_brk 2  sys_brk 3  syscall_return: Ok(Return(268439552))
syscall_number: 12  sys_brk 1  new heap end = 10001dc0sys_brk 2  sys_brk 3  syscall_return: Ok(Return(268443648))
syscall_number: 158  syscall_return: Ok(Return(0))
syscall_number: 218  syscall_return: Ok(Return(16))
syscall_number: 273  syscall_return: Ok(Return(0))
syscall_number: 334  syscall_return: Err(Error { errno: ENOSYS, msg: Some("Syscall was unimplemented") })
syscall_number: 63  syscall_return: Ok(Return(0))
syscall_number: 302  syscall_return: Ok(Return(0))
syscall_number: 89  syscall_return: Ok(Return(16))
syscall_number: 318  syscall_return: Ok(Return(8))
sys_wait4 8  syscall_return: Ok(Return(15))
syscall_number: 15  syscall_return: Ok(NoReturn)
syscall_number: 61  sys_wait4 1  syscall_number: 12  sys_brk 1  new heap end = 10023000sys_brk 2  Printing stack trace:


```

It seems that these logs do not print the stack trace successfully, and then hang.

**Comment by tatetian** [comment]
Could you provide a reliable and minimal way to reproduce the problem?

**Comment by tangruize** [comment]
#1540 might be related to this issue, which causes deadlock in `print_stack_trace()`.

**Comment by Hsy-Intel** [comment]
> #1540 might be related to this issue, which causes deadlock in `print_stack_trace()`.



> #1540 might be related to this issue, which causes deadlock in `print_stack_trace()`.

It looks consistent with the hang log in this question, but we may need to further confirm the root cause of the panic.

**Comment by Hsy-Intel** [comment]
This issue is resolved by #1637.
